src/pages/ResourceView.tsx

```diff
@@ -1,28 +1,152 @@
     setResource({ ...resource, ...updatedResourceData.resource })
   }
+  const queryClient = useQueryClient()
+
+  const [commentText, setCommentText] = useState('')
+
+  const [submitDisabled, setSubmitDisabled] = useState(false)
+
+  useEffect(() => setSubmitDisabled(commentText == ''), [commentText])
+
+  const onSubmit = async (data: IResourceCommentParams) => {
+    setSubmitDisabled(true)
+    const commentSuccess = await ApiResources.createComment(resource.id, data)
+
+    if (commentSuccess) {
+      toast.success(t('resources.comments.createdSuccess'))
+      setCommentText('')
+      queryClient.invalidateQueries({
+        queryKey: ['resourceComments', resource.id],
+      })
+    } else {
+      toast.error(t('error.generic'))
+    }
+
+    setSubmitDisabled(false)
+  }
+
+  const { data, fetchNextPage, hasNextPage, isFetching } = useInfiniteQuery({
+    queryKey: ['resourceComments', resource.id],
+    queryFn: async ({ pageParam = 0 }) =>
+      await ApiResources.listComments(resource.id, { page: pageParam }),
+    getNextPageParam: ({ meta }) =>
+      meta.page < meta.totalPages - 1 ? meta.page + 1 : undefined,
+    initialPageParam: 0,
+  })
+
+  const queryData = data || { pages: [] }
 
   return (
-    <div className="vertical-rhythm">
-      <div>
-        <a role="button" onClick={() => navigate(-1)}>
-          <i className="bi bi-chevron-left align-middle" />
-          {t('ui.back')}
-        </a>
+    <>
+      <PageHeader title={''} showBack />
+      <div className="vertical-rhythm">
+        <ResourceCard resource={resource} onUpdate={onUpdateResource} />
+        <Divider />
+        <div className="vertical-rhythm-sm">
+          {user ? (
+            <Card body>
+              <div className="position-relative">
+                <FormControl
+                  className="bg-light border-0 shadow-none"
+                  as="textarea"
+                  value={commentText}
+                  onChange={(e) => setCommentText(e.target.value)}
+                  placeholder={t('resources.comments.add')}
+                  rows={1}
+                />
+                <Button
+                  variant="link"
+                  className="p-0 text-dark position-absolute d-flex align-items-center"
+                  style={{
+                    right: '1rem',
+                    bottom: '0',
+                    height: '38px',
+                  }}
+                  disabled={submitDisabled}
+                  onClick={() => onSubmit({ body: commentText })}
+                >
+                  <SendIcon
+                    fill={
+                      submitDisabled
+                        ? bootstrapVariables['mediumGray']
+                        : bootstrapVariables['cobalt']
+                    }
+                  />
+                </Button>
+              </div>
+            </Card>
+          ) : (
+            <>
+              <Button
+                className="w-100"
```

Comment:
The full-width looks a bit awkward on desktop:
<img width="1131" alt="image" src="https://github.com/ProjectProtocol/project-protocol-web/assets/6488787/0d3cd890-060d-4f44-86d7-9dc05aa628a7">

You can use screensize modifiers on the classes so they apply for different screen sizes. Here you can do `w-100 w-md-auto` on the button, which will make it 100% width until medium+ screen sizes (medium is generally our break point between mobile and desktop styles).

Reply:
oh cool! good call

Reply:
thanks! seems like i had forgotten to add `d-flex` the most important className